app/slashcommands/command_templates.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package slashcommands
+
+import (
+	"github.com/mattermost/mattermost-server/v6/app"
+	"github.com/mattermost/mattermost-server/v6/app/request"
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/i18n"
+)
+
+type TemplatesProvider struct {
+}
+
+const (
+	CmdTemplates = "templates"
+)
+
+func init() {
+	app.RegisterCommandProvider(&TemplatesProvider{})
+}
+
+func (h *TemplatesProvider) GetTrigger() string {
+	return CmdTemplates
+}
+
+func (h *TemplatesProvider) GetCommand(a *app.App, T i18n.TranslateFunc) *model.Command {
+	workTemplateEnabled := a.Config().FeatureFlags.WorkTemplate
+	pbActive, err := a.IsPluginActive(model.PluginIdPlaybooks)
+	if err != nil {
+		pbActive = false
+	}
+	hasBoard, err := a.HasBoardProduct()
+	if err != nil {
+		hasBoard = false
+	}
+
+	return &model.Command{
+		Trigger:          CmdTemplates,
+		AutoComplete:     hasBoard && pbActive && workTemplateEnabled,
```

Comment:
Will this not list it in autocomplete if the products are not installed or FF is not enabled? But will it still open if the user does the slash command ?

Reply:
On the server side this controls if `/templates` is suggested or not as the user type in the webapp. If the user on the webapp decides to type by themselves `/templates`, we also have a control there to prevent the modal from opening:

https://github.com/mattermost/mattermost-webapp/pull/12145/files#diff-90e355879fe49fceb37601ed999e0a25253bab68d463a024c73ee7bf406182bbR145-R146

Reply:
Perfect, thanks!